setting DEIS_TEST_APP causes tests to fail

It seems if `DEIS_TEST_APP` is set to something other than the default, the integration tests will fail.
```console
$ DEIS_TEST_APP=example-ruby-sinatra ./tests/bin/test-integration.sh
...
=== configsample
BUILDPACK_URL: https://github.com/heroku/heroku-buildpack-go#98f37cc
POWERED_BY: the Deis team
ok
git push deis master
Warning: Permanently added '[deis.local3.deisapp.com]:2222,[172.17.8.102]:2222' (RSA) to the list of known hosts.
-----> Fetching custom buildpack
-----> no app detected
 !     A .godir is required. For instructions:
 !     http://mmcgrana.github.io/2012/09/getting-started-with-go-on-heroku
To ssh://git@deis.local3.deisapp.com:2222/configsample.git
 * [new branch]      master -> master
ok
--- FAIL: TestConfig (28.52s)
	itutils.go:139: App not started (503)
		Body: (<html>
		<head><title>503 Service Temporarily Unavailable</title></head>
		<body bgcolor="white">
		<center><h1>503 Service Temporarily Unavailable</h1></center>
		<hr><center>nginx/1.6.2</center>
		</body>
		</html>
		)
FAIL
```
We should skip that section of the config test if `DEIS_TEST_APP` isn't what we expect, or else create similar tests for the other example apps.

See also #2879.

--- a/tests/config_test.go
+++ b/tests/config_test.go
@@ -4,6 +4,7 @@ package tests
 
 import (
 	"io/ioutil"
+	"strings"
 	"testing"
 
 	"github.com/deis/deis/tests/utils"
@@ -14,9 +15,23 @@ var (
 	configSetCmd          = "config:set FOO=讲台 --app={{.AppName}}"
 	configSet2Cmd         = "config:set FOO=10 --app={{.AppName}}"
 	configSet3Cmd         = "config:set POWERED_BY=\"the Deis team\" --app={{.AppName}}"
-	configSetBuildpackCmd = "config:set BUILDPACK_URL=https://github.com/heroku/heroku-buildpack-go#98f37cc --app={{.AppName}}"
+	configSetBuildpackCmd = "config:set BUILDPACK_URL=$BUILDPACK_URL --app={{.AppName}}"
 	configUnsetCmd        = "config:unset FOO --app={{.AppName}}"
 )
+
+var buildpacks = map[string]string{
+	"example-clojure-ring":   "https://github.com/heroku/heroku-buildpack-clojure#v66",
+	"example-go":             "https://github.com/heroku/heroku-buildpack-go#6eeb09f",
+	"example-java-jetty":     "https://github.com/heroku/heroku-buildpack-java#v38",
+	"example-nodejs-express": "https://github.com/heroku/heroku-buildpack-nodejs#v75",
+	"example-perl":           "https://github.com/miyagawa/heroku-buildpack-perl#2da7480",
+	"example-php":            "https://github.com/heroku/heroku-buildpack-php#v67",
+	"example-play":           "https://github.com/heroku/heroku-buildpack-play#v23",
+	"example-python-django":  "https://github.com/heroku/heroku-buildpack-python#v58",
+	"example-python-flask":   "https://github.com/heroku/heroku-buildpack-python#v58",
+	"example-ruby-sinatra":   "https://github.com/heroku/heroku-buildpack-ruby#v137",
+	"example-scala":          "https://github.com/heroku/heroku-buildpack-scala#v55",
+}
 
 func TestConfig(t *testing.T) {
 	params := configSetup(t)
@@ -47,8 +62,14 @@ func configSetup(t *testing.T) *utils.DeisTestConfig {
 	// ensure envvars with spaces work fine on `git push`
 	// https://github.com/deis/deis/issues/2477
 	utils.Execute(t, configSet3Cmd, cfg, false, "the Deis team")
-	// ensure custom buildpack URLS are in order
-	utils.Execute(t, configSetBuildpackCmd, cfg, false, "https://github.com/heroku/heroku-buildpack-go#98f37cc")
+	// ensure custom buildpack URLs are in order
+	url := buildpacks[cfg.ExampleApp]
+	if url == "" {
+		// set url anyway so example-dockerfile apps create a build
+		url = buildpacks["example-go"]
+	}
+	cmd := strings.Replace(configSetBuildpackCmd, "$BUILDPACK_URL", url, 1)
+	utils.Execute(t, cmd, cfg, false, url)
 	utils.Execute(t, gitPushCmd, cfg, false, "")
 	utils.CurlApp(t, *cfg)
 	utils.CheckList(t, "run env --app={{.AppName}}", cfg, "DEIS_APP", false)

